ios/Classes/components/applePay/ApplePaySessionComponent.swift

```diff
@@ -31,14 +28,13 @@ class ApplePaySessionComponent: BaseApplePayComponent {
         applePayComponent = nil
     }
     
-    private func buildApplePaySessionComponent() -> ApplePayComponent? {
-        guard let session = sessionHolder.session,
-              let paymentMethod = session.sessionContext.paymentMethods.paymentMethod(ofType: ApplePayPaymentMethod.self),
-              let applePayComponent = try? ApplePayComponent(paymentMethod: paymentMethod, context: adyenContext, configuration: configuration)
-        else {
-            return nil
-        }
-        applePayComponent.delegate = sessionHolder.session
+    private func buildApplePaySessionComponent() throws -> ApplePayComponent? {
+        guard let session = sessionHolder.session else { throw PlatformError(errorDescription: "Session is not available.") }
+        guard let paymentMethod = session.sessionContext.paymentMethods.paymentMethod(ofType: ApplePayPaymentMethod.self) else { throw PlatformError(errorDescription: "Apple Pay payment method not valid.") }
+        let context = try configuration.createAdyenContext()
+        let configuration = try configuration.mapToApplePayConfiguration(payment: session.sessionContext.payment ?? context.payment)
+        let applePayComponent = try? ApplePayComponent(paymentMethod: paymentMethod, context: context, configuration: configuration)
```

Comment:
Is it still valid that this try does not throw?

Reply:
Good catch!